internal/linker/link_node_modules.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @fileoverview Creates a node_modules directory in the current working directory
+ * and symlinks in the node modules needed to run a program.
+ * This replaces the need for custom module resolution logic inside the process.
+ */
+const fs = require('fs');
+const path = require('path');
+
+function debug(...m) {
+  if (!!process.env['VERBOSE_LOGS']) console.error('[link_node_modules.js]', ...m);
+}
+
+function ls_r(d) {
+  for (const p of fs.readdirSync(d)) {
+    const f = path.join(d, p);
+    debug('  ', f);
+    if (fs.statSync(f).isDirectory()) {
+      ls_r(f);
+    }
+  }
+}
+
+function symlink(target, path) {
+  debug(`symlink( ${path} -> ${target} )`);
+  fs.symlinkSync(target, path);
```

Comment:
that's what the logic does now, but we do it before calling the symlink helper.
I guess it would be more clear if the short-circuit just happens in the helper instead